test/analysis/weightFunction.test.ts

```diff
@@ -0,0 +1,102 @@
+import 'ts-jest';
+import { buildSchema, DocumentNode, parse } from 'graphql';
+import { TypeWeightObject } from '../../src/@types/buildTypeWeights';
+import buildTypeWeightsFromSchema from '../../src/analysis/buildTypeWeights';
+import getQueryTypeComplexity from '../../src/analysis/typeComplexityAnalysis';
+// Test the weight function generated by the typeweights object when a limiting keyword is provided
+
+// Test cases:
+// Default value provided to schema
+// Arg passed in as variable
+// Arg passed in as scalar
+// Invalid arg type provided
+
+// Default value passed with query
+
+describe('Weight Function correctly parses Argument Nodes if', () => {
+    const schema = buildSchema(`
+        type Query {
+            reviews(episode: Episode!, first: Int = 5): [Review]
+            heroes(episode: Episode!, first: Int): [Review]
+            villains(episode: Episode!, limit: Int! = 3): [Review] 
+            characters(episode: Episode!, limit: Int!): [Review] 
+        }
+        type Review {
+            episode: Episode
+            stars: Int!
+            commentary: String
+        }
+        enum Episode {
+            NEWHOPE
+            EMPIRE
+            JEDI
+        }`);
+
+    // building the typeWeights object here since we're testing the weight function created in
+    // the typeWeights object
+    const typeWeights: TypeWeightObject = buildTypeWeightsFromSchema(schema);
+
+    describe('a default value is provided in the schema', () => {
+        xtest('and a value is not provided with the query', () => {
+            const query = `query { reviews(episode: NEWHOPE) { stars, episode } }`;
+            const queryAST: DocumentNode = parse(query);
+            expect(getQueryTypeComplexity(queryAST, {}, typeWeights)).toBe(6);
+        });
+
+        test('and a scalar value is provided with the query', () => {
+            const query = `query { reviews(episode: NEWHOPE, first: 3) { stars, episode } }`;
+            const queryAST: DocumentNode = parse(query);
+            expect(getQueryTypeComplexity(queryAST, {}, typeWeights)).toBe(4);
+        });
+
+        xtest('and the argument is passed in as a variable', () => {
+            const query = `query variableQuery ($items: Int){ reviews(episode: NEWHOPE, first: $items) { stars, episode } }`;
+            const queryAST: DocumentNode = parse(query);
+            expect(getQueryTypeComplexity(queryAST, { items: 7, first: 4 }, typeWeights)).toBe(9);
```

Comment:
Why `items` and ` first` as variables? The query only uses `items` to resolve reviews and I would expect 8 (7 reviews + 1 query)